src/test/java/com/vinyl/GetCartDetailsTest.java

```diff
@@ -24,16 +26,26 @@
     @Autowired
     TokenHeaderHelper tokenHeaderHelper;
 
+    @Autowired
+    ProductCartRepository productCartRepository;
+
+    @Autowired
+    CartRepository cartRepository;
+
     @Test
     public void testWhenUserLoggedIn() {
         Product product = defaultEntitiesHelper.createProduct();
         Cart cart =defaultEntitiesHelper.createCart(user);
         ProductCart pc=defaultEntitiesHelper.createProductCart(cart, product);
         cart.setProducts(Lists.newArrayList(pc));
+        cartRepository.save(cart);
 
-        ResponseEntity<?> cdo=setUpHeaderAndGetTheResponse();
+        ResponseEntity<CartDetailsDTO> cdo=setUpHeaderAndGetTheResponse();
 
         Assertions.assertThat(cdo.getStatusCode()).isEqualTo(HttpStatus.OK);
+        Assertions.assertThat(cdo.getBody().getNrProducts()).isEqualTo(productCartRepository.findByCart(cart).size());
+        Assertions.assertThat(cdo.getBody().getProducts().contains(pc));
```

Comment:
hmmm.. does this assert work? it should not, you are directly comparing a ProductDTO to a ProductCart....

Reply:
it does work only because i put the brackets wrong. so the assert only checks if there is a result between brackets, whatever it is. i'll fix this